webapp/src/globalContext/GlobalContext.tsx

```diff
@@ -0,0 +1,40 @@
+import { createProvider } from 'tg.fixtures/createProvider';
+import { useOrganizationUsageService } from './useOrganizationUsageService';
+import { useInitialDataService } from './useInitialDataService';
+import { GlobalActionType, globalDispatchRef } from './globalActions';
+
+export const [GlobalProvider, useGlobalDispatch, useGlobalContext] =
+  createProvider(() => {
+    const initialData = useInitialDataService();
+
+    const organizationUsage = useOrganizationUsageService({
+      organization: initialData.data.preferredOrganization,
+      enabled: Boolean(initialData.data?.serverConfiguration?.billing.enabled),
+    });
+
+    const dispatch = async (action: GlobalActionType) => {
+      switch (action.type) {
+        case 'UPDATE_ORGANIZATION':
+          return initialData.updatePreferredOrganization(action.payload);
+        case 'REFETCH_INITIAL_DATA':
+          return initialData.refetchInitialData();
+        case 'REFETCH_USAGE':
+          return organizationUsage.refetch();
+        case 'UPDATE_USAGE':
+          return organizationUsage.updateData(action.payload);
+        case 'INCREMENT_PLAN_LIMIT_ERRORS':
+          return organizationUsage.incrementPlanLimitErrors();
+      }
+    };
+
+    globalDispatchRef.current = dispatch;
+
+    const contextData = {
+      ...initialData.data!,
+      isFetching: initialData.isFetching,
+      isLoading: initialData.isLoading,
+      organizationUsage: organizationUsage.data,
+    };
+
+    return [contextData, dispatch];
```

Comment:
Why can't we just export the methods and data in an object without the Redux approach? Then you woudn't have to define the type separately. Also also it would imporve the readibility of the code. In my opinion. Every time I see. ```something.dispatch({type: 'DO_SOMETHING''})``` It takes me like 30 seconds to understand what happening. ```something.doSomething()``` is pretty clear right away.

Reply:
⬆️